Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeInteractionRegionLayers.mm

```diff
@@ -40,56 +40,54 @@
 static void configureLayerForInteractionRegion(CALayer *, NSString *) { }
 #endif
 
-@interface WKInteractionRegion : NSObject
-@property (nonatomic, assign) WebCore::InteractionRegion interactionRegion;
-@end
-
-@implementation WKInteractionRegion
-@end
-
 namespace WebKit {
 using namespace WebCore;
 
-NSString *interactionRegionKey = @"WKInteractionRegion";
-NSString *interactionRegionOcclusionKey = @"WKInteractionRegionOcclusion";
+NSString *interactionRegionTypeKey = @"WKInteractionRegionType";
+NSString *interactionRegionGroupNameKey = @"WKInteractionRegionGroupName";
+
+static std::optional<WebCore::InteractionRegion::Type> interactionRegionTypeForLayer(CALayer *layer)
+{
+    id value = [layer valueForKey:interactionRegionTypeKey];
+    if (value)
+        return static_cast<InteractionRegion::Type>([value boolValue]);
+    return std::nullopt;
+}
+
+static NSString * interactionRegionGroupNameForLayer(CALayer *layer)
+{
+    return [layer valueForKey:interactionRegionGroupNameKey];
+}
 
-static std::optional<WebCore::InteractionRegion> interactionRegionForLayer(CALayer *layer)
+static NSString* interactionRegionGroupNameForRegion(const WebCore::InteractionRegion& interactionRegion)
 {
-    id value = [layer valueForKey:interactionRegionKey];
-    if (![value isKindOfClass:[WKInteractionRegion class]])
-        return std::nullopt;
-    WKInteractionRegion *region = (WKInteractionRegion *)value;
-    return region.interactionRegion;
+    return makeString("WKInteractionRegion-"_s, interactionRegion.elementIdentifier.toUInt64());
 }
 
 static bool isInteractionLayer(CALayer *layer)
 {
-    return !!interactionRegionForLayer(layer);
+    return interactionRegionTypeForLayer(layer) == InteractionRegion::Type::Interaction;
 }
 
 static bool isOcclusionLayer(CALayer *layer)
 {
-    id value = [layer valueForKey:interactionRegionOcclusionKey];
-    if (value)
-        return true;
-    return false;
+    return interactionRegionTypeForLayer(layer) == InteractionRegion::Type::Occlusion;
 }
 
 static bool isAnyInteractionRegionLayer(CALayer *layer)
 {
-    return isOcclusionLayer(layer) || isInteractionLayer(layer);
+    return !!interactionRegionTypeForLayer(layer);
 }
 
-static void setInteractionRegion(CALayer *layer, const WebCore::InteractionRegion& interactionRegion)
+static void setInteractionRegion(CALayer *layer, NSString *groupName)
 {
-    WKInteractionRegion *region = [[[WKInteractionRegion alloc] init] autorelease];
-    region.interactionRegion = interactionRegion;
-    [layer setValue:region forKey:interactionRegionKey];
+    [layer setValue:@(static_cast<bool>(InteractionRegion::Type::Interaction)) forKey:interactionRegionTypeKey];
+    [layer setValue:groupName forKey:interactionRegionGroupNameKey];
```

Comment:
Secretly I wonder how much we'd win if we used our own hash table instead of KVC